lib/Mojo/Promise.pm

```diff
@@ -288,6 +310,40 @@ rejects. If the returned promise fulfills, it is fulfilled with the values from
 the fulfilled promises in the same order as the passed promises. This method can
 be useful for aggregating results of multiple promises.
 
+=head2 all_settled
+
+  # create two pending promises
+  my $promise_1 = Mojo::Promise->new(sub ($resolve, $reject) {
+    Mojo::IOLoop->timer(0.5 => sub { $resolve->(3) });
+  });
+  my $promise_2 = Mojo::Promise->new(sub ($resolve, $reject) {
+    Mojo::IOLoop->timer(0.1 => sub { $reject->('foo') });
+  });
+
+  # create new promise
+  my $new = Mojo::Promise->all_settled($promise_1, $promise_2);
+
+  # check the status of promises
+  $new->then(sub (@results) {
+    for my $result (@results) {
+      say "$result->{status}: $result->{value}[0]"
+        if $result->{status} eq 'fulfilled';
+      say "$result->{status}: $result->{reason}[0]"
+        if $result->{status} eq 'rejected';
+    }
+  })->wait;
+
+Returns a new L<Mojo::Promise> object that fulfills when all of the
+passed L<Mojo::Promise> objects have either fulfilled or rejected.
+
```

Comment:
```pod
Returns a new L<Mojo::Promise> object that fulfills when all of the
supplied L<Mojo::Promise> objects have settled as I<fulfilled> or I<rejected>.
```

Reply:
is "fulfills" the preferred term here? I thought with promises, they are either *resolved* or rejected.

Reply:
Yes, It's part of the standard. See [mdn link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled).